Update core eggs to new docker yolk images

notAreYouScared opened the pull request:
**W.I.P**

Upgrading core eggs to use the new yolk images.


- [X] Minecraft
- [X] ~~Rust~~ Furture PR
- [x] Source Games
  - [X] ~~Ark~~ Furture PR
  - [x] CS:GO
  - [x] Custom Source Engine Game
  - [x] GMod
  - [x] Insurgency
  - [x] TF2
- [x] Voice Servers

notAreYouScared committed:
Update eslint to latest verstions

Updated Eslint to v7.27.0
Also updated all the related plugins to their latest versions.

Removed `eslint-plugin-standard`, no longer required with `eslint-config-standard` v16+   https://github.com/standard/standard/issues/1316

notAreYouScared committed:
Update Minecraft eggs to use yolks

notAreYouScared committed:
Change order of minecraft java versions

notAreYouScared committed:
Update minecraft install images, update rust

notAreYouScared committed:
Merge branch 'develop' into update/eggs

notAreYouScared committed:
Update source eggs

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -4,29 +4,39 @@
+    "exported_at": "2021-06-05T15:47:39-04:00",
     "name": "Ark: Survival Evolved",
-    "author": "dev@shepper.fr",
+    "author": "support@pterodactyl.io",
```
Why remove the credit of the original author?

notAreYouScared replied:
That's what was in the egg when I exported it, as we found out I had an old version of the egg. Will be updated.

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -40,46 +50,28 @@
-            "description": "ARK server name",
-            "env_variable": "SESSION_NAME",
-            "default_value": "A Pterodactyl Hosted ARK Server",
+            "name": "Server Port",
```
This is the primary port, you don't have to create a variable for this.

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -40,46 +50,28 @@
+            "description": "Enable or disable rcon system. (true or false)",
             "env_variable": "ENABLE_RCON",
-            "default_value": "True",
+            "default_value": "false",
```
RCON is required for all console output and disabling it by default will flood Discord channels complaining why rcon fails to connect

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -40,46 +50,28 @@
-            "user_editable": false,
-            "rules": "required|string|in:True,False"
+            "user_editable": true,
+            "rules": "required|string|max:5"
```
Why remove the existing validation rule `in:True,False` which allows only True or False?

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -100,22 +92,31 @@
             "rules": "required|numeric"
         },
         {
-            "name": "Auto-update server",
```
Why remove this option when it exists and is useful, because breaking the whole server with outdated mods is not ideal

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -100,22 +92,31 @@
+            "rules": "required|numeric|digits_between:1,4"
         },
         {
-            "name": "Battle Eye",
```
Why remove this option?

trenutoo commented on database/Seeders/eggs/source-engine/egg-insurgency.json:
```diff
@@ -1,24 +1,29 @@
         "files": "{}",
-        "startup": "{\"done\": \"gameserver Steam ID\", \"userInteraction\": []}",
-        "logs": "{\"custom\": true, \"location\": \"logs\/latest.log\"}",
+        "startup": "{\r\n    \"done\": \"gameserver Steam ID\",\r\n    \"userInteraction\": []\r\n}",
```
userInteraction doesn't exist anymore, it is replaced by the `features` of the egg. Could remove it since you are updating all eggs at once

notAreYouScared replied:
Removed userInteraction, was still present in the egg shipped with the panel.

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -4,29 +4,39 @@
     "config": {
         "files": "{}",
-        "startup": "{\r\n    \"done\": \"Waiting commands for 127.0.0.1:\",\r\n    \"userInteraction\": []\r\n}",
+        "startup": "{\r\n    \"done\": \"Setting breakpad minidump AppID = 346110\",\r\n    \"userInteraction\": []\r\n}",
```
This is no indication of a successful server startup, it will always show when you try to start the server even if it gets stuck or fails to start. You only see the minidump because you disabled RCON for console output

trenutoo commented on database/Seeders/eggs/source-engine/egg-ark--survival-evolved.json:
```diff
@@ -4,29 +4,39 @@
     ],
-    "startup": "rmv() { echo -e \"stoppping server\"; rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} -c saveworld && rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} -c DoExit; }; trap rmv 15; cd ShooterGame\/Binaries\/Linux && .\/ShooterGameServer {{SERVER_MAP}}?listen?SessionName=\"{{SESSION_NAME}}\"?ServerPassword={{ARK_PASSWORD}}?ServerAdminPassword={{ARK_ADMIN_PASSWOR ...
+    "file_denylist": [],
+    "startup": "\"\/home\/container\/ShooterGame\/Binaries\/Linux\/ShooterGameServer {{SERVER_MAP}}?listen?SessionName='{{SESSION_NAME}}'?ServerPassword={{ARK_PASSWORD}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}?Port={{PORT}}?MaxPlayers={{SERVER_MAX_PLAYERS}}?RCONPort={{RCON_PORT}}?QueryPort={{QUERY_PORT}}?RCONEnabled={{ENABLE_RCON}} -server -log\"",
```
Your startup does not work at this moment as it's enclosed in double-quotes, also you removed the server save before shutdown that the previous trap did

trenutoo reviewed (commented):
Not sure why this PR is not just for changing images themselves instead of modifying the eggs with some drastic breaking changes for ARK as an example

notAreYouScared committed:
Merge branch 'develop' into update/eggs

notAreYouScared committed:
revert rust

Changes will come in another pr

notAreYouScared committed:
Update

notAreYouScared committed:
Merge branch 'update/eggs' of https://github.com/AreYouRlyScared/panel into update/eggs

notAreYouScared committed:
revert ark changes

an update will come in a later pr

notAreYouScared committed:
Fix Mumble

notAreYouScared committed:
Fix ts3 egg

this is dependent on https://github.com/pterodactyl/yolks/pull/5

DaneEveritt committed:
Merge branch 'develop' into update/eggs

DaneEveritt commented:
@AreYouRlyScared to confirm, this is ready to go now?

I removed all the java images except 8, 11, and 16. Just want to make sure nothing was impacted by that.

notAreYouScared committed:
Update Eggs

Removed J9/J11 images from list.
Removed old `UserInteraction` from startup config

notAreYouScared commented:
> @AreYouRlyScared to confirm, this is ready to go now?
> 
> I removed all the java images except 8, 11, and 16. Just want to make sure nothing was impacted by that.

@DaneEveritt  All Set. Removed J9 / J14 ( ik the pr says J9 + J11, but It was J14 that was removed ). Also removed the old userinteraction as they're also not used.

notAreYouScared committed:
Remove deprecated UserInteraction from tests and egg config

Removes deprecated UserInteraction from test and egg config service
